src/assignment/views/AssignmentEdit.tsx

```diff
@@ -89,131 +88,111 @@ const AssignmentEdit: FunctionComponent<DefaultSecureRouteProps<{ id: string }>>
+	const checkPermissionsAndGetAssignment = useCallback(async () => {
+		try {
+			// Redirect if id is a legacy numeric assignment id instead of a guid
+			const assignmentId = match.params.id;
+
+			if (AssignmentService.isLegacyAssignmentId(assignmentId)) {
+				const assignmentUuid:
+					| string
+					| undefined = await AssignmentService.getAssignmentUuidFromLegacyId(
+					assignmentId
+				);
+				if (!assignmentUuid) {
+					console.error(
+						new CustomError(
+							'The assignment id appears to be a legacy assignment id, but the matching uuid could not be found in the database',
+							null,
+							{ legacyId: assignmentId }
+						)
 					);
+					setLoadingInfo({
+						state: 'error',
+						message: t(
+							'assignment/views/assignment-edit___de-opdracht-kon-niet-worden-gevonden'
+						),
+						icon: 'search',
+					});
 					return;
 				}
+				history.replace(buildLink(APP_PATH.ASSIGNMENT_EDIT.route, { id: assignmentUuid }));
+				return;
+			}
 
-				// Determine if this is an edit or create page and initialize or fetch the assignment
-				const tempAssignment: Partial<
-					Avo.Assignment.Assignment
-				> | null = await fetchAssignment(assignmentId);
-
-				if (!tempAssignment) {
-					// Something went wrong during init/fetch
-					return;
-				}
-
-				// Fetch the content if the assignment has content
-				let tempAssignmentContent: AssignmentContent | null = null;
-				try {
-					tempAssignmentContent = await AssignmentService.fetchAssignmentContent(
-						tempAssignment
-					);
-
-					setAssignmentContent(tempAssignmentContent);
-				} catch (err) {
-					if (err.message !== 'NOT_FOUND') {
-						console.error(
-							new CustomError('Failed to fetch assignment content', err, {
-								assignment: tempAssignment,
-							})
-						);
-						ToastService.danger(
-							t(
-								'assignment/views/assignment-edit___het-ophalen-van-de-opdracht-inhoud-is-mislukt'
-							)
-						);
-					}
+			// Determine if this is an edit or create page and initialize or fetch the assignment
+			const tempAssignment: Partial<
+				Avo.Assignment.Assignment
+			> | null = await AssignmentService.fetchAssignmentByUuid(assignmentId);
 
-					setAssignmentContent(null);
-				}
+			if (!tempAssignment) {
+				// Something went wrong during init/fetch
+				return;
+			}
 
-				setBothAssignments({
-					...tempAssignment,
-					title: tempAssignment.title || get(tempAssignmentContent, 'title', ''),
-				});
-				setAssignmentLabels(
-					AssignmentLabelsService.getLabelsFromAssignment(tempAssignment)
+			if (
+				!(await PermissionService.hasPermissions(
+					[
+						PermissionName.EDIT_ALL_ASSIGNMENTS,
+						{ name: PermissionName.EDIT_ASSIGNMENTS, obj: tempAssignment },
+						{ name: PermissionName.EDIT_OWN_ASSIGNMENTS, obj: tempAssignment },
+					],
+					user
+				))
+			) {
+				history.push(`/${ROUTE_PARTS.assignments}/${assignmentId}`);
+				ToastService.danger(
```

Comment:
ToastService.info seems more appropriate. Since we mostly use danger for stuff related to deletes

Reply:
updated.